### PR TITLE
Testing task implementation

### PR DESCRIPTION
**Pull request by OMManko**
e2e test is located in cypress/integration/e2e.spec.js

### CODE REVIEW

File: src/components/infoPanel/infoPanel.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            <InfoPanel />
+        );
+    });
+
+    afterEach(() => {
+        component.unmount();
+    });
+
+    it('should render info panel and match snapshot', () => {
+        component = shallow(<InfoPanel/>);
```
Comment (nsasinovich) [inline src/components/infoPanel/infoPanel.test.js]: Already in beforeEach

File: src/components/layout/header/header.test.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { shallow } from 'enzyme';
+import React from 'react';
+import Header from '../header/header';
+import AppLogo from "../../shared/appLogo/appLogo";
+
+const component = shallow(<Header/>);
```
Comment (nsasinovich) [inline src/components/layout/header/header.test.js]: move to beforeEach hook please

File: src/components/layout/main/main.test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            <Main />
+        );
+    });
+
+    afterEach(() => {
+        component.unmount();
+    });
+
+    it('should render main part and match snapshot', () => {
+        component = shallow(<Main/>);
```
Comment (nsasinovich) [inline src/components/layout/main/main.test.js]: already in beforeEach

File: src/components/movieCard/movieCard.test.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { mount, shallow } from 'enzyme';
+import React from 'react';
+import MovieCard from './movieCard';
+import MoviePoster from "../moviePoster/moviePoster";
+
+let component;
+
+describe('<MovieCard />', () => {
+    it('should render radio group and match snapshot', () => {
+        component = shallow(<MovieCard />);
```
Comment (nsasinovich) [inline src/components/movieCard/movieCard.test.js]: to beforeEach

File: src/components/movieInfo/movieInfo.jsx [hunk 5]
```diff
@@ -5,28 +5,23 @@ import PropTypes from "prop-types";
 import MoviePoster from "../moviePoster/moviePoster";
 import { moviePosterSizes } from '../../constants/constants';
 
-const MovieInfo = ({ src, title, genre, year, duration, description }) => (
+const MovieInfo = ({ movie }) => (
     <article className="movieInfo">
-        <MoviePoster src={src} size={moviePosterSizes.SMALL}/>
+        <MoviePoster src={movie.src} size={moviePosterSizes.SMALL}/>
```
Comment (nsasinovich) [inline src/components/movieInfo/movieInfo.jsx]: destructuring from movie?

File: src/components/searchResults/searchResults.test.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            <SearchResults />
+        );
+    });
+
+    afterEach(() => {
+        component.unmount();
+    });
+
+    it('should render search results block and match snapshot', () => {
+        component = shallow(<SearchResults/>);
```
Comment (nsasinovich) [inline src/components/searchResults/searchResults.test.js]: in beforeEach

File: src/components/searchResults/searchResults.test.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        expect(component).toMatchSnapshot();
+    });
+
+    it('should render no results block', () => {
+        expect(component.find(NoResults).length).toEqual(1);
+    });
+
+    it('should render movie list', () => {
+        expect(component.find(MovieList).length).toEqual(1);
```
Comment (nsasinovich) [inline src/components/searchResults/searchResults.test.js]: or can be like this:
` expect(component.contains(<MovieList />)).toBe(true);`

File: src/components/infoPanel/infoPanel.test.js (hunk 1 above)
Comment (OMManko) [reply src/components/infoPanel/infoPanel.test.js]: Refactored

File: src/components/layout/main/main.test.js (hunk 3 above)
Comment (OMManko) [reply src/components/layout/main/main.test.js]: Refactored

File: src/components/searchResults/searchResults.test.js (hunk 6 above)
Comment (OMManko) [reply src/components/searchResults/searchResults.test.js]: Refactored

File: src/components/searchResults/searchResults.test.js (hunk 7 above)
Comment (OMManko) [reply src/components/searchResults/searchResults.test.js]: I leave as it is to show unit tests knowledge

File: src/components/movieCard/movieCard.test.js (hunk 4 above)
Comment (OMManko) [reply src/components/movieCard/movieCard.test.js]: Refactored

File: src/components/layout/header/header.test.js (hunk 2 above)
Comment (OMManko) [reply src/components/layout/header/header.test.js]: Refactored

**Pull request merged**